File: binance4j-core/src/main/java/com/binance4j/core/dto/RateLimit.java
package com.binance4j.core.dto;

import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;

/**
 * Requests and orders rate limit.
 * 
 * @param rateLimitType Rate limit type.
 * @param interval      Limit interval type.
 * @param intervalNum   Limit interval number.
 * @param limit         Limit for the given interval.
 */
@ApiModel("Requests and orders rate limit.")
public record RateLimit(@ApiModelProperty("The rate limit type.") String rateLimitType,
		@ApiModelProperty("The limit interval type.") String interval,
		@ApiModelProperty("The limit interval number.") int intervalNum,
		@ApiModelProperty("The limit for the given interval.") int limit) {
}

File: connectors/src/main/kotlin/com/binance4j/core/dto/RateLimit.kt
/*
 * MIT License
 *
 * Copyright (c) 2022 Binance4j
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NON INFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package com.binance4j.core.dto

import io.swagger.annotations.ApiModel
import io.swagger.annotations.ApiModelProperty

/**
 * Requests and orders rate limit.
 *
 * @property rateLimitType Rate limit type.
 * @property interval Limit interval type.
 * @property intervalNum Limit interval number.
 * @property limit Limit for the given interval.
 */
@ApiModel("Requests and orders rate limit.")
data class RateLimit(
	@ApiModelProperty("The rate limit type.")
	val rateLimitType: String,
	@ApiModelProperty("The limit interval type.")
	val interval: String,
	@ApiModelProperty("The limit interval number.")
	val intervalNum: Int,
	@ApiModelProperty("The limit for the given interval.")
	val limit: Int
)